### DISCUSSION TITLE
Welcome to fashion-mnist Discussions!

### DISCUSSION THREAD

**Discussion by kashif** [zalandoresearch/fashion-mnist#173] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Mita23456** [comment]
Hello

**Comment by hnqh888** [comment]
when I use the following code to download the fashion mnist dataset , it reports the following type error:
i use the tf-gpu virtual environment ,the tensorflow-datasets=4.9.2,tensorflow-gpu=2.3.0,the same problem occurs in MNIST datasets, i download the dataset file from tensorflow-datasets relative GitHub site and extracted normally.

code:
import tensorflow_datasets as tfds

data_dir = 'F:/TensorFlow/Tensorflow-vioce/fashiong mnist datasets'  
(train_data, test_data), metadata = tfds.load('fashion_mnist', 
                                              split=['train', 'test'], 
                                              data_dir=data_dir,
                                              download=True,
                                              as_supervised=True,
                                              with_info=True)

result:
TypeError                                 Traceback (most recent call last)
Cell In[2], line 4
      1 import tensorflow_datasets as tfds
      3 data_dir = 'F:/TensorFlow/Tensorflow-vioce/fashiong mnist datasets'  # 指定数据存储目录
----> 4 (train_data, test_data), metadata = tfds.load('fashion_mnist', 
      5                                               split=['train', 'test'], 
      6                                               data_dir=data_dir,
      7                                               download=True,
      8                                               as_supervised=True,
      9                                               with_info=True)

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\logging\__init__.py:169, in _FunctionDecorator.__call__(self, function, instance, args, kwargs)
    167 metadata = self._start_call()
    168 try:
--> 169   return function(*args, **kwargs)
    170 except Exception:
    171   metadata.mark_error()

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\load.py:640, in load(name, split, data_dir, batch_size, shuffle_files, download, as_supervised, decoders, read_config, with_info, builder_kwargs, download_and_prepare_kwargs, as_dataset_kwargs, try_gcs)
    521 """Loads the named dataset into a `tf.data.Dataset`.
    522 
    523 `tfds.load` is a convenience method that:
   (...)
    632     Split-specific information is available in `ds_info.splits`.
    633 """
    634 dbuilder = _fetch_builder(
    635     name,
    636     data_dir,
    637     builder_kwargs,
    638     try_gcs,
    639 )
--> 640 _download_and_prepare_builder(dbuilder, download, download_and_prepare_kwargs)
    642 if as_dataset_kwargs is None:
    643   as_dataset_kwargs = {}

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\load.py:499, in _download_and_prepare_builder(dbuilder, download, download_and_prepare_kwargs)
    497 if download:
    498   download_and_prepare_kwargs = download_and_prepare_kwargs or {}
--> 499   dbuilder.download_and_prepare(**download_and_prepare_kwargs)

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\logging\__init__.py:169, in _FunctionDecorator.__call__(self, function, instance, args, kwargs)
    167 metadata = self._start_call()
    168 try:
--> 169   return function(*args, **kwargs)
    170 except Exception:
    171   metadata.mark_error()

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\dataset_builder.py:646, in DatasetBuilder.download_and_prepare(self, download_dir, download_config, file_format)
    644   self.info.read_from_directory(self._data_dir)
    645 else:
--> 646   self._download_and_prepare(
    647       dl_manager=dl_manager,
    648       download_config=download_config,
    649   )
    651   # NOTE: If modifying the lines below to put additional information in
    652   # DatasetInfo, you'll likely also want to update
    653   # DatasetInfo.read_from_directory to possibly restore these attributes
    654   # when reading from package data.
    655   self.info.download_size = dl_manager.downloaded_size

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\dataset_builder.py:1535, in GeneratorBasedBuilder._download_and_prepare(self, dl_manager, download_config)
   1523   for split_name, generator in utils.tqdm(
   1524       split_generators.items(),
   1525       desc="Generating splits...",
   1526       unit=" splits",
   1527       leave=False,
   1528   ):
   1529     filename_template = naming.ShardedFileTemplate(
   1530         split=split_name,
   1531         dataset_name=self.name,
   1532         data_dir=self.data_path,
   1533         filetype_suffix=path_suffix,
   1534     )
-> 1535     future = split_builder.submit_split_generation(
   1536         split_name=split_name,
   1537         generator=generator,
   1538         filename_template=filename_template,
   1539         disable_shuffling=self.info.disable_shuffling,
   1540     )
   1541     split_info_futures.append(future)
   1543 # Process the result of the beam pipeline.

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\split_builder.py:341, in SplitBuilder.submit_split_generation(self, split_name, generator, filename_template, disable_shuffling)
    338 # Depending on the type of generator, we use the corresponding
    339 # `_build_from_xyz` method.
    340 if isinstance(generator, collections.abc.Iterable):
--> 341   return self._build_from_generator(**build_kwargs)
    342 else:  # Otherwise, beam required
    343   unknown_generator_type = TypeError(
    344       f'Invalid split generator value for split `{split_name}`. '
    345       'Expected generator or apache_beam object. Got: '
    346       f'{type(generator)}'
    347   )

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\core\split_builder.py:406, in SplitBuilder._build_from_generator(self, split_name, generator, filename_template, disable_shuffling)
    396 serialized_info = self._features.get_serialized_info()
    397 writer = writer_lib.Writer(
    398     serializer=example_serializer.ExampleSerializer(serialized_info),
    399     filename_template=filename_template,
   (...)
    404     shard_config=self._shard_config,
    405 )
--> 406 for key, example in utils.tqdm(
    407     generator,
    408     desc=f'Generating {split_name} examples...',
    409     unit=' examples',
    410     total=total_num_examples,
    411     leave=False,
    412 ):
    413   try:
    414     example = self._features.encode_example(example)

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tqdm\std.py:1182, in tqdm.__iter__(self)
   1179 time = self._time
   1181 try:
-> 1182     for obj in iterable:
   1183         yield obj
   1184         # Update and possibly print the progressbar.
   1185         # Note: does not call self.update(1) for speed optimisation.

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\image_classification\mnist.py:156, in MNIST._generate_examples(self, num_examples, data_path, label_path)
    145 """Generate MNIST examples as dicts.
    146 
    147 Args:
   (...)
    153   Generator yielding the next examples
    154 """
    155 images = _extract_mnist_images(data_path, num_examples)
--> 156 labels = _extract_mnist_labels(label_path, num_examples)
    157 data = list(zip(images, labels))
    159 # Using index as key since data is always loaded in same order.

File D:\anaconda3\envs\tf-gpu\lib\site-packages\tensorflow_datasets\image_classification\mnist.py:393, in _extract_mnist_labels(labels_filepath, num_labels)
    391 def _extract_mnist_labels(labels_filepath, num_labels):
    392     with tf.io.gfile.GFile(labels_filepath, "rb") as f:
--> 393         f.read(8)  # header
    394         buf = f.read(num_labels)
    395         labels = np.frombuffer(buf, dtype=np.uint8).astype(np.int64)

File ~\AppData\Roaming\Python\Python38\site-packages\tensorflow\python\lib\io\file_io.py:116, in FileIO.read(self, n)
    104 def read(self, n=-1):
    105   """Returns the contents of a file as a string.
    106 
    107   Starts reading from current position in file.
   (...)
    114     string if in string (regular) mode.
    115   """
--> 116   self._preread_check()
    117   if n == -1:
    118     length = self.size() - self.tell()

File ~\AppData\Roaming\Python\Python38\site-packages\tensorflow\python\lib\io\file_io.py:79, in FileIO._preread_check(self)
     76   raise errors.PermissionDeniedError(None, None,
     77                                      "File isn't open for reading")
     78 print("self.__name:",self.__name)
---> 79 self._read_buf = _pywrap_file_io.BufferedInputStream(self.__name, 1024 * 1024)

TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. tensorflow.python._pywrap_file_io.BufferedInputStream(arg0: str, arg1: int)

Invoked with: WindowsGPath('F:\\TensorFlow\\Tensorflow-vioce\\fashiong mnist datasets\\downloads\\extracted\\GZIP.fashion-mnist_train-labels-idx1-ubyteoE8XE0rANWCkfjdk4RuS_JfeTRv6-LoaOqKa9UzJCEU.gz'), 1048576

**Comment by Felipe-Ordenes** [comment]
Hola, no puedo descargar la data
